### ISSUE TITLE
Authorization header should begin with `Signature`

### ISSUE THREAD

**Issue by ewilde** [go-fed/httpsig#3]
Try to validate signatures generated by `httpsig`, with other languages i.e.: https://github.com/joyent/node-http-signature.

It appears that `httpsig` does not add the `Signature` value at the beginning of the header. According to the [Signing HTTP Messages draft-cavage-http-signatures-10](https://tools.ietf.org/html/draft-cavage-http-signatures-10#section-3.1.2)

**Draft example**
```
 Authorization: Signature keyId="rsa-key-1",algorithm="rsa-sha256",
   headers="(request-target) host date digest content-length",
   signature="Base64(RSA-SHA256(signing string))"
```

**Http sig**
```
Authorization: keyId="pubKeyId",algorithm="rsa-sha512",headers="date digest",signature="Base64(RSA-SHA256(signing string))"
```

**Comment by cjslep** [comment]
Thanks, I am embarassed to admit I totally forgot about the `Authorization` header's `auth-scheme`. Fixed in both signing and verifying in c86d639563fbc8f31c10e52f61ebacd05b74a95c

**Comment by cjslep** [comment]
I modernized the test to use the `t.Run` subtests with table tests.

Also noticed I did not add a unit test case enforcing the authScheme.

Added that test. Result: I had only fixed it for non-MAC based algorithms. So now this bug is wholly fixed (2555707168f018a8d68061077712b5f20d06029b), and has test coverage.
